src/Maestro/Maestro.Web/Api/v2019_01_16/Controllers/GoalController.cs

```diff
@@ -0,0 +1,93 @@
+        [HttpPut]
+        [SwaggerApiResponse(System.Net.HttpStatusCode.OK, Type = typeof(Models.Goal), Description = "Sets a build time goal (in minutes) for a given Definition in a Channel.")]
+        [ValidateModelState]
+        public virtual async Task<IActionResult> Create([FromBody, Required] Goal.GoalData goalData,[Required] String channelName , [Required] int definitionId)
+        {
+            Data.Models.Channel channel = await _context.Channels
+                .Where(c => c.Name.Equals(channelName)).FirstOrDefaultAsync();
+            if (channel == null)
+            {
+                return NotFound();
+            }
+            Data.Models.GoalTime goal = await _context.GoalTime
+                .Where(g => g.DefinitionId == definitionId && g.ChannelId == channel.Id).FirstOrDefaultAsync();
+            if (goal == null)
+            {
+                goal = new Data.Models.GoalTime
+                {
+                    DefinitionId = definitionId,
+                    Minutes = goalData.Minutes,
+                    ChannelId = channel.Id
+                };
+                await _context.GoalTime.AddAsync(goal);
+            }
+            else
+            // If the combination of Channel and DefinitionId already exists then update the exisiting record
+            {
+                goal.Minutes = goalData.Minutes;
+                _context.GoalTime.Update(goal);
+            }
+            await _context.SaveChangesAsync();
+            return Ok(new Goal(goal));
+        }
+
+        /// <summary>
+        /// Gets the build time in minutes <see cref="Goal"/> for a given Definition in a Channel. This is captured for the Power BI Dashboard -Internal Report under .Net Core Engineering Services workspace.
+        /// </summary>
+        /// <param name="channelName">Channel Name for the build time Eg. .Net Core 5</param>
+        /// <param name="definitionId">Azure DevOps pipeline Definition Id</param>
+        [HttpGet("channelName/{channelName}/definitionId/{definitionId}")]
+        [SwaggerApiResponse(HttpStatusCode.OK, Type = typeof(Goal), Description = "Gets the build time goal (in minutes) for a given Definition in a Channel.")]
+        [ValidateModelState]
+        public virtual async Task<IActionResult> GetGoalTimes([Required]int definitionId, [Required]string channelName)
+        {
+            Data.Models.Channel channel = await _context.Channels
+                .Where(c => c.Name.Equals(channelName)).FirstOrDefaultAsync();
+            if (channel == null)
+            {
+                return NotFound();
+            }
+            Data.Models.GoalTime goal = await _context.GoalTime
+                .Where(g => g.DefinitionId == definitionId && g.ChannelId == channel.Id).FirstOrDefaultAsync();
+            if (goal == null)
+            {
+                return NotFound();
+            }
+            return Ok(goal);
```

Comment:
>goal [](start = 22, length = 4)

This is not returning the right type. You need to construct a "Goal" object (the one in the v2019_01_16/Models folder) from the GoalTime value.

Reply:
returning new goal object